File: src/components/pages/projects/$projectId/shared/connect-snippets/android/java/MainActivity.java
import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;

import io.appwrite.coroutines.CoroutineCallback;
import io.appwrite.services.Account;

public class MainActivity extends Activity {
  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    showHome();
  }

  void showHome() {
    setContentView(createHome());
  }

  void showSignIn() {
    setContentView(SignIn.create(this, this::showHome, this::showSignUp));
  }

  void showSignUp() {
    setContentView(SignUp.create(this, this::showHome, this::showSignIn));
  }

  private View createHome() {
    LinearLayout layout = new LinearLayout(this);
    layout.setOrientation(LinearLayout.VERTICAL);
    int pad = (int) (24 * getResources().getDisplayMetrics().density);
    layout.setPadding(pad, pad, pad, pad);

    TextView status = new TextView(this);
    status.setText("Loading...");
    layout.addView(status);

    Account account = new Account(AppwriteClient.get(this));
    try {
      account.get(new CoroutineCallback<>((user, error) ->
          layout.post(() -> {
            layout.removeAllViews();
            if (user == null) {
              addSignedOutHome(layout);
            } else {
              addSignedInHome(layout, user.getName());
            }
          })));
    } catch (Exception ignored) {}
    return layout;
  }

  private void addSignedOutHome(LinearLayout layout) {
    TextView cta = new TextView(this);
    cta.setText("Sign in to get started.");
    layout.addView(cta);

    Button signIn = new Button(this);
    signIn.setText("Sign in");
    signIn.setAllCaps(false);
    signIn.setOnClickListener(v -> showSignIn());
    layout.addView(signIn);

    Button signUp = new Button(this);
    signUp.setText("Sign up");
    signUp.setAllCaps(false);
    signUp.setOnClickListener(v -> showSignUp());
    layout.addView(signUp);
  }

  private void addSignedInHome(LinearLayout layout, String name) {
    TextView greeting = new TextView(this);
    greeting.setText("Hello, " + name);
    layout.addView(greeting);

    Button signOut = new Button(this);
    signOut.setText("Sign out");
    signOut.setAllCaps(false);
    signOut.setOnClickListener(v -> {
      Account account = new Account(AppwriteClient.get(this));
      try {
        account.deleteSession("current",
            new CoroutineCallback<>((result, error) ->
                layout.post(() -> {
                  if (error == null) showHome();
                })));
      } catch (Exception ignored) {}
    });
    layout.addView(signOut);
  }
}

File: src/components/pages/projects/$projectId/shared/connect-snippets/android/kotlin/MainActivity.kt
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.unit.dp
import io.appwrite.services.Account
import kotlinx.coroutines.launch

class MainActivity : ComponentActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContent { App() }
  }
}

@Composable
fun App() {
  var route by remember { mutableStateOf("home") }

  Column(modifier = Modifier.padding(24.dp)) {
    when (route) {
      "sign-in" -> SignIn(
        onSignedIn = { route = "home" },
        onGoToSignUp = { route = "sign-up" },
      )
      "sign-up" -> SignUp(
        onSignedUp = { route = "home" },
        onGoToSignIn = { route = "sign-in" },
      )
      else -> Home(
        onGoToSignIn = { route = "sign-in" },
        onGoToSignUp = { route = "sign-up" },
      )
    }
  }
}

@Composable
fun Home(onGoToSignIn: () -> Unit, onGoToSignUp: () -> Unit) {
  val context = LocalContext.current
  val scope = rememberCoroutineScope()
  var name by remember { mutableStateOf<String?>(null) }
  var loading by remember { mutableStateOf(true) }

  LaunchedEffect(Unit) {
    name = try {
      Account(AppwriteClient.get(context)).get().name
    } catch (e: Exception) {
      null
    }
    loading = false
  }

  if (loading) {
    Text("Loading...")
    return
  }

  if (name == null) {
    Text("Sign in to get started.")
    TextButton(onClick = onGoToSignIn) { Text("Sign in") }
    TextButton(onClick = onGoToSignUp) { Text("Sign up") }
    return
  }

  Text("Hello, $name")
  TextButton(
    onClick = {
      scope.launch {
        Account(AppwriteClient.get(context)).deleteSession("current")
        name = null
      }
    },
  ) { Text("Sign out") }
}
